src/bluesky/run_engine.py

```diff
@@ -197,7 +198,12 @@ def __get__(self, instance, owner):
             return super().__get__(instance, owner)
 
 
-def default_scan_id_source(md):
+# type alias
+# TODO(kivel): check with core devs where to put this
+RunEngineMetadata = MutableMapping[str, typing.Any]
```

Comment:
I think here is fine.  Makes sense to me to import this from the same place you import RunEngine from.